### ISSUE TITLE
Protoc --ruby_out fails to generate a file with a message that contains a field of type google.protobuf.DescriptorProto

### ISSUE THREAD

**Issue by viacheslav-rostovtsev** [protocolbuffers/protobuf#7992]
**What version of protobuf and what language are you using?**
Version: 3.13.0
Language: Ruby

**What operating system (Linux, Windows, ...) and version?**
Debian 5.7.17-1rodete3 (2020-09-16) x86_64 GNU/Linux  

**What runtime / compiler are you using (e.g., python version or gcc version)**
ruby 2.6.5

**What did you do?**
1. Checkout `https://github.com/googleapis/googleapis/blob/master/google/cloud/bigquery/storage/v1beta2/protobuf.proto`
(( the message that cause an issue is:
```
message ProtoSchema {
  // Descriptor for input message. The descriptor has to be self contained,
  // including all the nested types, excepted for proto buffer well known types
  // (https://developers.google.com/protocol-buffers/docs/reference/google.protobuf).
  google.protobuf.DescriptorProto proto_descriptor = 1;
}
```

2. run `protoc --ruby_out=/tmp/foo protobuf.proto`
3. see error `--ruby_out: protobuf.proto: proto3 message field google.cloud.bigquery.storage.v1beta2.ProtoSchema.proto_descriptor in file google/protobuf/descriptor.proto has a dependency on a type from proto2 file google/protobuf/descriptor.proto.  Ruby doesn't support proto2 yet, so we must fail.`

**What did you expect to see**
ruby code generated at the target directory

**What did you see instead?**
an error
--ruby_out: protobuf.proto: proto3 message field google.cloud.bigquery.storage.v1beta2.ProtoSchema.proto_descriptor in file google/protobuf/descriptor.proto has a dependency on a type from proto2 file google/protobuf/descriptor.proto.  Ruby doesn't support proto2 yet, so we must fail.`

**Anything else we should know about your project / environment**
-

**Comment by yirutang** [comment]
Any updates on this? Thanks.

**Comment by zhangskz** [comment]
The fix was merged to branch 3.18.x on Wednesday and should be part of the next release.
